source/val/validate_interfaces.cpp

```diff
@@ -529,6 +529,61 @@ spv_result_t GetLocationsForVariable(
   return SPV_SUCCESS;
 }
 
+bool IsInterpolationDecoration(spv::Decoration dec) {
+  return dec == spv::Decoration::PerVertexKHR || dec == spv::Decoration::Flat ||
+         dec == spv::Decoration::NoPerspective ||
+         dec == spv::Decoration::Sample || dec == spv::Decoration::Centroid;
+}
+
+// Need to check Variables (or struct members) with explicit Locaiton decoration
+// don't mix interpolation decorations
+spv_result_t ValidateLocationInterpolation(
+    ValidationState_t& _, bool is_input, const Instruction* variable,
+    std::unordered_map<uint32_t, std::vector<spv::Decoration>>&
+        interpolation_map) {
+  uint32_t location = 0;
+  bool has_location = false;
+  std::vector<spv::Decoration> interpolation_decs;
+  for (auto& dec : _.id_decorations(variable->id())) {
+    spv::Decoration decoration = dec.dec_type();
+    if (decoration == spv::Decoration::Location) {
+      location = dec.params()[0];
+      has_location = true;
+    } else if (IsInterpolationDecoration(decoration)) {
+      interpolation_decs.push_back(decoration);
+    }
+  }
+
+  // Look for Location in a Block decorated Struct
+  if (!has_location) {
+    auto ptr_type_id = variable->GetOperandAs<uint32_t>(0);
+    auto ptr_type = _.FindDef(ptr_type_id);
+    auto type_id = ptr_type->GetOperandAs<uint32_t>(2);
+    for (auto& dec : _.id_decorations(type_id)) {
+      spv::Decoration decoration = dec.dec_type();
+      if (decoration == spv::Decoration::Location) {
+        location = dec.params()[0];
+        has_location = true;
+      } else if (IsInterpolationDecoration(decoration)) {
+        interpolation_decs.push_back(decoration);
```

Comment:
This seems like it would allow:
```
OpMemberDecorate %struct 0 Location 0
OpMemberDecorate %struct 0 Sample
OpMemberDecorate %struct 0 Centroid
```
Is that caught elsewhere? Certainly it should be invalid.